src/elements/hx-file-input/index.spec.js

```diff
@@ -0,0 +1,121 @@
+
+                const shadow = component.shadowRoot.innerHTML;
+
+                expect(component).shadowDom.to.equal(shadow);
+            });
+
+            it('should render the Shadow Root mode open', async () => {
+                const component = /** @type { HXFileInputElement } */ await fixture(template);
+                const mode = component.shadowRoot.mode;
+
+                expect(mode).to.equal("open");
+            });
+
+            it('should have a single <slot>', async () => {
+                const component = /** @type { HXFileInputElement } */ await fixture(template);
+                const shadow = component.shadowRoot;
+                const query = shadow.querySelectorAll('slot');
+                const len = query.length;
+
+                expect(len).to.be.equal(1);
+            });
+
+            it('should have an unnamed <slot>', async () => {
+                const component = /** @type { HXFileInputElement } */ await fixture(template);
+                const name = component.slot;
+
+                if ( name !== null ) {
+                    expect(name).to.be.equal('');
+                } else {
+                    expect(name).to.be.null;  // IE11, Legacy Edge, and older browsers
+                }
+            });
+        });
+
+        describe('verify Shadow DOM markup', () => {
+            it('markup should contain a #hxFileInput <div>', async () => {
+                const elSelector = 'div#hxFileInput';
+                const id = 'hxFileInput';
+                const component = /** @type { HXFileInputElement } */ await fixture(template);
+                const shadow = component.shadowRoot;
+                const query = shadow.querySelector(elSelector);
+                const queryId = query.id;
+
+                expect(queryId).to.equal(id);
+            });
+
+            it('markup should contain a #hxIcon', async () => {
+                const elSelector = 'div#hxFileInput > #hxIcon';
+                const id = 'hxIcon';
+                const component = /** @type { HXFileInputElement } */ await fixture(template);
+                const shadow = component.shadowRoot;
+                const query = shadow.querySelector(elSelector);
+                const queryId = query.id;
+                expect(queryId).to.equal(id);
+            });
+
+            it('markup should contain a #hxLabel <span>', async () => {
+                const elSelector = 'span#hxLabel';
+                const id = 'hxLabel';
+                const component = /** @type { HXFileInputElement } */ await fixture(template);
+                const shadow = component.shadowRoot;
+                const query = shadow.querySelector(elSelector);
+                const queryId = query.id;
+
+                expect(queryId).to.equal(id);
+            });
+        });
+    });
+
+    describe('test getters and setters', () => {
+        it('should be able to set icon [icon="upload"]', async () => {
```

Comment:
You are on the right path.  Please update this test to set/check this `[icon="upload"]`.

Reply:
Changes have been done, Please review